akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -84,6 +84,45 @@ class FusingSpec extends StreamSpec {
       refs.toSet should have size (in.size + 1) // outer/main actor + 1 actor per subflow
     }
 
+    "propagate downstream errors through async boundary" in {
+      val slowInitSrc = Source
+      //todo: can this be replaced with some promise+future composition?
+        .unfoldResource(() => { Thread.sleep(800) }, (_: Unit) => Some(1), (_: Unit) => ())
+        .watchTermination()(Keep.right)
+        .async //commenting this out, makes the test pass
+      val downstream = Flow[Int]
+        .prepend(Source.single(1))
+        .flatMapPrefix(0) {
+          case Nil => sys.error("I hate mondays")
```

Comment:
Use `TE` or `TestException` instead of generic exception, always good when looking at logs to discern exceptions that were on purpose.

Reply:
👍